spec/request/api/v1/auth/sign_up_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+describe 'Auth', type: :request do
+  let(:user) { build(:user) }
+  let(:params) do
+    {
+      name: user.name,
+      email: user.email,
+      password: user.password,
+      password_confirmation: user.password
+    }
+  end
+
+  describe 'POST api/v1/auth' do
+    describe 'success' do
+      it 'should respond with 200 OK' do
+        post api_v1_user_registration_path params
+        expect(response).to be_successful
+      end
+
+      it 'should increase user count by 1' do
+        expect do
+          post api_v1_user_registration_path params
+        end.to change(User, :count).by(1)
+      end
+
+      it 'should contains headers #access-token #client #expiry #uid' do
+        post api_v1_user_registration_path params
+        expect(response.headers['access-token']).not_to be_nil
+        expect(response.headers['client']).not_to be_nil
+        expect(response.headers['expiry']).not_to be_nil
+        expect(response.headers['uid']).to eq user.email
+      end
+
+      it 'should contains body data' do
+        post api_v1_user_registration_path params
+        body = JSON response.body
+        expect(body['status']).to eq 'success'
+        expect(body['data']['id']).not_to be_nil
```

Comment:
This is fine, but it would be nice to add a helper in order to avoid the need of `body['data']['id']`. Maybe a method like `def body(name)` that returns either the property (if given) or the whole body otherwise (just if needed).